tests/rptest/services/admin.py

```diff
@@ -521,15 +522,24 @@ def _request(self,
         if kwargs.get('timeout', None) is None:
             kwargs['timeout'] = DEFAULT_TIMEOUT
 
+        #We will have to handle redirects ourselves (always set kwargs['allow_redirects'] = False),
+        #see comment after _session.request() below.
+        #If kwargs was passed with "allow_redirects:False", it is assumed that the user intends
+        #to handle the redirect case at the call site. Otherwise, it will be retried in the
+        #request loop below.
+        handle_retry_backoff = kwargs.get('allow_redirects', True)
+        kwargs['allow_redirects'] = False
```

Comment:
Yes, we do. The `kwargs` variable is going to be passed to `self._session.request()` on L545, which [uses the `allow_redirects`](https://requests.readthedocs.io/en/latest/api/#requests.request) value to determine whether it should internally retry on a 30X error code or not.

We don't want this to happen, as it will not respect a `Retry-After` header for timeouts, as I explain in the comment on L558-563, 

```
# Requests library does NOT respect Retry-After with a 307
# error code (see https://github.com/psf/requests/pull/4562).
# There is logic that respects Retry-After within urllib3,
# but since the Requests library handles 30# error codes
# internally, a Retry-After attached to a redirect response
# will not be respected. We will have to handle this ourselves.
```

This leads into the logic on L564-573 for manually respecting the `Retry-After`.

Please let me know if this all makes sense.

Reply:
It does but then the condition in L564 means that redirects are only implemented for 307 error code but there are other 3xx redirect error codes that are not handled or am i reading it incorrectly?

Reply:
Correct, currently this is only going to handle `307`s, since that is what `redirect_to_leader()` is throwing from `server.cc`. But, for complete coverage, maybe it should be checking `300 <= r.status_code < 400`.

Reply:
Changing this line to 

`if handle_retry_backoff and **r.is_redirect** and num_redirects < self._session.max_redirects:`

Reply:
cool that makes sense.